.circleci/examples.py

```diff
@@ -52,9 +55,35 @@ def divide_list(lst, n):
 # Returns a 2-elem tuple, the first element is the list of special examples, the second is
 #   a list of lists of regular jobs, divided into <parallel_jobs - special_jobs> chunks
 #   (i.e. divided evenly into remaining parallel jobs)
+@lru_cache
 def split_examples_into_jobs(connector):
   special_examples = get_special_for(connector)
   regular_examples = get_regular_for(connector)
   parallel_jobs = parallel_jobs_for(connector)
   regular_example_jobs = parallel_jobs - len(special_examples)
   return (special_examples, divide_list(regular_examples, regular_example_jobs))
+
+# Decide which examples to run on the current runner using `circleci tests split`
+#
+# This passes every special example on its own line,
+# and "job-group-<n>" for every set of regular examples.
+# Then, this decodes the line printed by the split command into a list
+# of examples to run, and returns that list.
+def circleci_split_examples(connector):
+  (special_examples, regular_examples) = split_examples_into_jobs(connector)
+  split_args = special_examples + [f"job-group-{n}" for n in range(len(regular_examples))]
+  assert len(split_args) == int(os.environ["CIRCLE_NODE_TOTAL"])
+  split_args_bytes = "\n".join(split_args).encode('utf-8')
+  circleci_split = subprocess.run(["circleci", "tests", "split"], input=split_args_bytes,
```

Comment:
How come you're doing it the hard way with this extra process and text processing? I think you should do it via just selecting the right element of the return of `split_examples_into_jobs` by looking at `CIRCLE_NODE_INDEX`

Reply:
I didn't think of doing it that way

Reply:
I mentioned it in an earlier comment, I guess it got delete.